### PR TITLE
VBMS efolder pagination

### LINKED ISSUE #13047: Fix VBMS::DocumentTooBig errors
https://github.com/department-of-veterans-affairs/connect_vbms/pull/241 implements a new PagedDocuments VBMS service.

### AC
- [ ] Upgrade connect_vbms gem
- [ ] Use the new service in our `VBMSService` class instead of `VBMS::Requests::FindDocumentVersionReference`

### Testing

Test in prod with PID 3978641 and any others.

### Related

#10665 
#10901 
https://github.com/department-of-veterans-affairs/caseflow-efolder/issues/1160

### PR DESCRIPTION
**Pull request by pkarman**
Resolves #13047 

### Description
Updates VBMS and BGS gems.

Adds feature toggle `:vbms_pagination` to control whether we use the new VBMS paged documents service to allow fetching VBMS efolders where there are 5k+ documents.

### Testing Plan
1. Identify a Veteran with 5k+ efolder in UAT (https://dsva.slack.com/archives/CK466V65V/p1577992111002300?thread_ts=1577990211.002000&cid=CK466V65V)
2. In Rails console, toggle the `:vbms_pagination` feature for yourself.
3. In Rails console, call the `ExternalApi::VbmsDocumentsForAppeal` for the Veteran.
4. In Rails console, call the `ExternalApi::VBMSService.fetch_document_series_for` method for the Veteran.

Repeat in production after deployment.

### CODE REVIEW

File: .reek.yml [hunk 1]
```diff
@@ -16,11 +16,13 @@ detectors:
       - Veteran
   ControlParameter:
     exclude:
+      - ExternalApi::VBMSService#self.send_and_log_request
```
Comment (pkarman) [inline .reek.yml]: existing violations

File: spec/services/vbms_documents_for_appeal_spec.rb [hunk 2]
```diff
@@ -91,4 +91,59 @@
...
+      it "looks up the BGS claim number in VBMS" do
+        bgs_claim_number = "87654321"
+
+        allow(vbms_client).to receive(:send_request).and_raise(nonexistent_file_number_error)
+        allow(bgs_client).to receive(:fetch_veteran_info).with(veteran_file_number)
+          .and_return(claim_number: bgs_claim_number)
+        allow(docs).to receive(:vbms_paged_documents_service) { pagination_service }
+
+        expect(vbms_client).to receive(:send_request).exactly(:twice)
+        expect { docs.fetch }.to raise_error(VBMS::FilenumberDoesNotExist)
```
Comment (leikkisa) [inline spec/services/vbms_documents_for_appeal_spec.rb]: Is this error getting raised because of line 139? It looks in the code like it raises this error if the claim number isn't different, but if it is different, it attempts to use `bgs_claim_number_docs_request`

File: spec/services/vbms_documents_for_appeal_spec.rb [hunk 3]
```diff
@@ -91,4 +91,59 @@
...
+
+        result_hash = {
+          manifest_vbms_fetched_at: nil,
+          manifest_vva_fetched_at: nil,
+          documents: docs_from_vbms_array
+        }
+
+        expect(bgs_client).to_not receive(:fetch_veteran_info)
+        expect(vbms_client).to_not receive(:send_request)
+        expect(docs.fetch).to eq result_hash
```
Comment (leikkisa) [inline spec/services/vbms_documents_for_appeal_spec.rb]: I'm having some trouble understanding this test.  If the pagination service is turned on, why is not returning the pagination response.  It looks like mocking `DocumentsFromVbmsDocuments` is overriding the result, independent of whether the pagination service or the other service is being called.

Comment (pkarman) [reply spec/services/vbms_documents_for_appeal_spec.rb]: The internal pagination response, mocked, is `documents: [1, 2, 3]`.

The internal `docs_from_vbms` object, mocked, converted that to `[4, 5, 6]`.

What we are testing is that we call the pagination service (mocked) internally, rather than the older non-pagination service. The mocks allow us to avoid making actual HTTP calls, and instead define known return values. The return values themselves are meaningless.

File: spec/services/vbms_documents_for_appeal_spec.rb (hunk 2 above)
Comment (pkarman) [reply spec/services/vbms_documents_for_appeal_spec.rb]: yes, line 139 mocks the exception.

The trigger for trying a 2nd time with the bgs number is the exception. If the 2 numbers are the same or bgs number is nil, the trigger is ignored and a second exception raised.

File: spec/services/vbms_documents_for_appeal_spec.rb (hunk 3 above)
Comment (pkarman) [reply spec/services/vbms_documents_for_appeal_spec.rb]: I have commented and renamed things to hopefully be more clear. I basically just started with the existing test above and then changed it to use the pagination service.

File: spec/services/vbms_documents_for_appeal_spec.rb (hunk 2 above)
Comment (pkarman) [reply spec/services/vbms_documents_for_appeal_spec.rb]: I realize in copy/pasting the existing test I failed to update to account for the different service call. I will fix.

Comment (pkarman) [reply spec/services/vbms_documents_for_appeal_spec.rb]: updated, with comments.

**Comment by pkarman** [comment]
thanks @leikkisa -- updated tests with comments to hopefully make them more clear. Mocks and injections are hard to follow, IME.

File: spec/services/vbms_documents_for_appeal_spec.rb (hunk 3 above)
Comment (leikkisa) [reply spec/services/vbms_documents_for_appeal_spec.rb]: Okay. This is still still confusing to me, but I see that it's still testing that the new pagination service is being called.
